src/circuit/gadget/ecc/chip/mul.rs

```diff
@@ -0,0 +1,426 @@
+
+                #[cfg(test)]
+                // Check that the correct multiple is obtained.
+                {
+                    use group::Curve;
+
+                    let base = base.point();
+                    let alpha = alpha
+                        .value()
+                        .map(|alpha| pallas::Scalar::from_bytes(&alpha.to_bytes()).unwrap());
+                    let real_mul = base.zip(alpha).map(|(base, alpha)| base * alpha);
+                    let result = result.point();
+
+                    if let (Some(real_mul), Some(result)) = (real_mul, result) {
+                        assert_eq!(real_mul.to_affine(), result);
+                    }
+                }
+
+                let zs = {
+                    let mut zs = vec![z_init];
+                    zs.extend_from_slice(&zs_incomplete_hi);
+                    zs.extend_from_slice(&zs_incomplete_lo);
+                    zs.extend_from_slice(&zs_complete);
+                    zs.extend_from_slice(&[z_0]);
+                    assert_eq!(zs.len(), pallas::Scalar::NUM_BITS as usize + 1);
+
+                    // This reverses zs to give us [z_0, z_1, ..., z_{254}, z_{255}].
+                    zs.reverse();
+                    zs
+                };
+
+                Ok((result, zs))
+            },
+        )?;
+
+        self.overflow_config
+            .overflow_check(layouter.namespace(|| "overflow check"), alpha, &zs)?;
+
+        Ok(result)
+    }
+
+    fn process_lsb(
+        &self,
+        region: &mut Region<'_, pallas::Base>,
+        offset: usize,
+        base: &EccPoint,
+        acc: EccPoint,
+        z_1: Z<pallas::Base>,
+        lsb: Option<bool>,
+    ) -> Result<(EccPoint, Z<pallas::Base>), Error> {
+        // Assign z_0 = 2⋅z_1 + k_0
+        let z_0 = {
+            let z_0_val = z_1.value().zip(lsb).map(|(z_1, lsb)| {
+                let lsb = pallas::Base::from_u64(lsb as u64);
+                z_1 * pallas::Base::from_u64(2) + lsb
+            });
+            let z_0_cell = region.assign_advice(
+                || "z_0",
+                self.complete_config.z_complete,
+                offset,
+                || z_0_val.ok_or(Error::SynthesisError),
+            )?;
+
+            // Check that z_0 was properly derived from z_1.
+            self.q_mul_z.enable(region, offset)?;
+
+            Z(CellValue::new(z_0_cell, z_0_val))
+        };
+
+        // If `lsb` is 0, return `Acc + (-P)`. If `lsb` is 1, simply return `Acc + 0`.
+        let x_p = if let Some(lsb) = lsb {
+            if !lsb {
+                base.x.value()
+            } else {
+                Some(pallas::Base::zero())
+            }
+        } else {
+            None
+        };
+        let y_p = if let Some(lsb) = lsb {
+            if !lsb {
+                base.y.value().map(|y_p| -y_p)
+            } else {
+                Some(pallas::Base::zero())
+            }
+        } else {
+            None
+        };
```

Comment:
Nothing is constraining that this value (witnessed by the prover) is the correct value -- which should be Acc - P or Acc depending on the value of `lsb`, so there's a missing gate here.